src/library/scala/util/Random.scala

```diff
@@ -127,9 +127,15 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    *  @since 2.8
    */
   def alphanumeric: Stream[Char] = {
+    // unused - kept to maintain binary compatibility with 2.11.x
     def isAlphaNum(c: Char) = (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')
 
-    Stream continually nextPrintableChar filter isAlphaNum
+    def nextAlphaNum(): Char = {
+      val chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
+      chars(self.nextInt(62))
+    }
```

Comment:
I would either `chars(self nextInt chars.length)` if I weren't paranoid about every opcode, or I would `"ABC..." charAt (self nextInt 62)` otherwise. Actually, `chars charAt (self nextInt chars.length)`.